.github/workflows/lint.yml

```diff
@@ -1,34 +0,0 @@
-name: Lint code
```

Comment:
These have been removed, but it's not clear that they've been replaced. Have they?

Reply:
The main ci workflow does everything now. This has the benefit of only requiring a single action, and only needing to set up the environment once. It runs fairly quickly (~3 minutes, most of which is spent initializing gradle and building java).

Reply:
We _could_ separate out building codegen as a separate job within in the same action, but let's see if the action run time becomes an issue first.

Reply:
I'm not really concerned about runtime, i'm more concerned about separation of concerns. It's useful to see at a glance what's failing and why without having to dig into some logs. This isn't a blocker though - we can make that change later if we want